output.go

```diff
@@ -86,15 +128,27 @@ func RemoveOutput(output io.Writer) {
 	}
 }
 
+// RemoveAllOutputs removes all previously added outputs
+func RemoveAllOutputs() {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	for _, p := range outputs {
+		delete(hookFnPointers, p)
+	}
+
+	outputs = make(map[io.Writer]uintptr)
+}
+
 // AddHook causes logging activity to invoke the given hook function
-func AddHook(fn func(*Entry), opts ...func(*Options)) {
+func AddHook(fn hookFn, opts ...func(*Options)) {
```

Comment:
If we switch this to a private interface, is that going to make it hard/impossible for people to add their own hooks?

Reply:
It shouldn't, as long as their function has the same signature as hookFn